packages/database/src/helpers/job.ts

```diff
@@ -232,13 +247,23 @@ function computePaidJobStatus(job: JobWithRelations): SokosumiJobStatus {
       switch (latestJobStatus.status) {
         case AgentJobStatus.COMPLETED:
           return SokosumiJobStatus.COMPLETED;
+        case AgentJobStatus.AWAITING_PAYMENT:
+          return SokosumiJobStatus.PROCESSING;
+        case AgentJobStatus.FAILED:
+          return SokosumiJobStatus.FAILED;
+        case AgentJobStatus.RUNNING:
+          return SokosumiJobStatus.PROCESSING;
+        case AgentJobStatus.AWAITING_INPUT:
+          return SokosumiJobStatus.INPUT_REQUIRED;
         default:
           return SokosumiJobStatus.RESULT_PENDING;
       }
     case OnChainJobStatus.FUNDS_WITHDRAWN:
       switch (latestJobStatus.status) {
         case AgentJobStatus.COMPLETED:
           return SokosumiJobStatus.COMPLETED;
+        case AgentJobStatus.AWAITING_PAYMENT:
+          return SokosumiJobStatus.PROCESSING;
```

Comment:
Why did you change the logic? I already found a mistake that the job with `id` is showing in your branch as working state, but it's actually result missing.